examples/platform/silabs/efr32/spi_multiplex.h

```diff
@@ -17,49 +17,152 @@
 
 #pragma once
 
+#if defined(EFR32MG24)
 #ifdef __cplusplus
 extern "C" {
 #endif
 #include "FreeRTOS.h"
 #include "semphr.h"
 #include "sl_mx25_flash_shutdown_usart_config.h"
+#if (defined(EFR32MG24) && defined(WF200_WIFI))
 #include "sl_spidrv_exp_config.h"
+#endif /* EFR32MG24 && WF200_WIFI */
+#if (defined(EFR32MG24) && defined(RS911X_WIFI))
+#include "sl_spidrv_eusart_exp_config.h"
+#endif /* EFR32MG24 && RS911X_WIFI */
+#if (defined(EFR32MG24) && defined(WF200_WIFI))
 #include "sl_wfx_host_api.h"
+#endif /* EFR32MG24 && WF200_WIFI */
+#include "em_eusart.h"
 #include "spidrv.h"
 
-#define SL_BIT_RATE_LCD 1100000
-#define SL_BIT_RATE_EXP_HDR 16000000
-#define SL_BIT_RATE_SPI_FLASH 10000000
-#define SL_BIT_RATE_UART_CONSOLE 16000000
+#ifdef WF200_WIFI
+// TODO: Investigate why using SL_SPIDRV_EXP_BITRATE is causing WF200 init failure
+// REF: sl_spidrv_exp_config.h
+#define SL_SPIDRV_EXP_BITRATE_MULTIPLEXED 10000000
+#else
+#define SL_SPIDRV_EXP_BITRATE_MULTIPLEXED SL_SPIDRV_EUSART_EXP_BITRATE
+#endif /* WF200_WIFI */
 
-extern SemaphoreHandle_t spi_sem_sync_hdl;
+#define SL_SPIDRV_LCD_BITRATE 1100000
+#define SL_SPIDRV_MX25_FLASH_BITRATE 16000000
+#define SL_SPIDRV_UART_CONSOLE_BITRATE 16000000
 
 #ifdef RS911X_WIFI
+#define SL_SPIDRV_FRAME_LENGTH SL_SPIDRV_EUSART_EXP_FRAME_LENGTH
 extern SPIDRV_Handle_t sl_spidrv_eusart_exp_handle;
 #define SL_SPIDRV_HANDLE sl_spidrv_eusart_exp_handle
-#endif
-
-#ifdef WF200_WIFI
+extern SPIDRV_Init_t sl_spidrv_eusart_init_exp;
+#elif WF200_WIFI
+#define SL_SPIDRV_FRAME_LENGTH SL_SPIDRV_EXP_FRAME_LENGTH
 extern SPIDRV_Handle_t sl_spidrv_exp_handle;
 #define SL_SPIDRV_HANDLE sl_spidrv_exp_handle
 #endif
 
-void spi_drv_reinit(uint32_t);
+/****************************************************************************
+ * @fn  void SPIDRV_ReInit()
+ * @brief
+ *     Re-Intializes SPI driver to required baudrate
+ * @param[in] None
+ * @return returns void
+ *****************************************************************************/
+void SPIDRV_ReInit(uint32_t);
```

Comment:
This seems bad. Why have a ReInit() function ? couldn't we just call the Init function again ?